src/backend/distributed/commands/index.c

```diff
@@ -151,6 +151,17 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand,
 	 * index statements.
 	 */
 	LOCKMODE lockMode = GetCreateIndexRelationLockMode(createIndexStatement);
+
+	/*
+	 * Acquire global lock to prevent concurrent writes or DDL. However,
+	 * we do not bother for CREATE INDEX CONCURRENTLY, since we'll have
+	 * to release the lock.
+	 */
+	if (!createIndexStatement->concurrent)
+	{
+		AcquireDistributedLockOnRelations(list_make1(relationRangeVar), lockMode, 0);
```

Comment:
Do we still have the issue you noted here: https://github.com/citusdata/citus/pull/5899#issuecomment-1099024350

My understanding at that time was that we want to fix this problem when we support DDL form workers. I feel like 11.1 could be a good time to have a robust implementation for that perspective (so that we can add other DDLs safer)

Mostly asking for clarification and reference. I think It is also fine to go with this patch for now (if we need CREATE INDEX in 11.1)

Reply:
I don't really know a good solution to this yet.